pkg/route/generator/generate.go

```diff
@@ -77,8 +78,14 @@ func (RouteGenerator) Generate(genericParams map[string]interface{}) (runtime.Ob
 
 	portString := params["target-port"]
 	if len(portString) > 0 {
+		var targetPort intstr.IntOrString
+		if port, err := strconv.Atoi(portString); err == nil {
+			targetPort = intstr.FromInt(port)
+		} else {
+			targetPort = intstr.FromString(portString)
+		}
 		route.Spec.Port = &api.RoutePort{
-			TargetPort: intstr.FromString(portString),
+			TargetPort: targetPort,
```

Comment:
So what I see is a check here just to invoke `intstr.From{Int,String}' and in the code further down (https://github.com/kargakis/origin/blob/route-status-fixes/pkg/route/graph/analysis/analysis.go#L80), there is still a call to the `intstr.String()` method and the `Atoi` conversion again [which is fine - we should check if its a string or an int at that point]. Seems like this conversion here is not really needed - just using `intstr.FromString()` would work here.